src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java

```diff
@@ -2,39 +2,64 @@
 
 import net.greatstart.dao.UserDao;
 import net.greatstart.model.User;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
 
 @Repository
 public class UserDaoImpl implements UserDao {
+    private SessionFactory sessionFactory;
+
+    @Autowired
+    public UserDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
     @Override
     public boolean create(User user) {
-        return false;
+        sessionFactory.getCurrentSession()
+                .save(user);
+        return true;
     }
 
     @Override
     public boolean delete(User user) {
-        return false;
+        sessionFactory.getCurrentSession()
+                .delete(user);
+        return true;
     }
 
     @Override
     public boolean update(User user) {
-        return false;
+        sessionFactory.getCurrentSession()
+                .update(user);
+        return true;
     }
 
     @Override
     public User getByEmail(String email) {
+        List<User> users = sessionFactory.getCurrentSession()
+                .createQuery("from User where email = :email", User.class)
+                .setParameter("email", email)
+                .getResultList();
+        if (users.size() > 0) {

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1155)